lib/ex_format/lines.ex

```diff
@@ -0,0 +1,15 @@
+defmodule ExFormat.Lines do
+  def initialize_lines_store(string) do
+  	lines = String.split(string, "\n")
+    Agent.start_link(fn -> %{} end, name: :lines)
+    for {line, i} <- Enum.with_index(lines) do
+      update_line(i + 1, String.trim(line))
+    end
```

Comment:
That's a great explanation. I have pushed a fix to stop the agent after formatting and it works. It indeed makes more sense to put everything in the formatter's state. I believe that requires a significant change which requires a separate PR. Are you alright with that?